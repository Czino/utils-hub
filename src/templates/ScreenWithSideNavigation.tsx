import { NavigationColumn } from './NavigationColumn'

export const ScreenWithSideNavigation = ({ children }: React.ComponentProps<'div'>) => (
  <div className="flex flex-row gap-4 text-2xl bg-white color-black">
    <NavigationColumn />
    <div className="p-4 space-y-4 w-full">{children}</div>
    <div className="w-[280px] hidden md:block">{/* placeholder */}</div>
  </div>
)
