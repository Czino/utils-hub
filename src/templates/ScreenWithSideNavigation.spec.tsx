import { render } from '@testing-library/react'
import { ScreenWithSideNavigation } from './ScreenWithSideNavigation'

describe('ScreenWithSideNavigation', () => {
  it('should render ScreenWithSideNavigation', () => {
    const { asFragment } = render(<ScreenWithSideNavigation />)
    expect(asFragment()).toMatchSnapshot()
  })
})
