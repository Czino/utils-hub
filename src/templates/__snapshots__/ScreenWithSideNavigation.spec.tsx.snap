// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenWithSideNavigation should render ScreenWithSideNavigation 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-4 text-2xl bg-white color-black"
  >
    <nav
      class="bg-white absolute p-4 transition-[top] w-full h-full top-full md:h-screen md:relative md:left-0 md:w-[240px]"
    >
      <ul>
        <li
          class="font-bold"
        >
          <a
            href="/"
          >
            UtilityHub
          </a>
        </li>
        <li>
          <a
            class=""
            href="/utils/list"
          >
            List
          </a>
        </li>
        <li>
          <a
            class=""
            href="/utils/text"
          >
            Text
          </a>
        </li>
        <li>
          <a
            class=""
            href="/utils/units"
          >
            Units
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="flex absolute right-0 bottom-0 justify-end p-4 w-full bg-white md:hidden"
      role="button"
    >
      <svg
        aria-label="open navigation"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="p-4 space-y-4 w-full"
    />
    <div
      class="w-[280px] hidden md:block"
    />
  </div>
</DocumentFragment>
`;
